src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -218,7 +218,7 @@ public ICompilationTracker FreezePartialStateWithTree(SolutionState solution, Do
                 // check whether we can bail out quickly for typing case
                 var inProgressState = state as InProgressState;
 
-                generatorInfo = state.GeneratorInfo;
+                generatorInfo = state.GeneratorInfo.WithDocumentsAreFinal(true);
```

Comment:
Just want to make sure, this is not part of the refactoring of passing the GeneratorInfo around, right? Looks like it's related to the actual fix?

Reply:
So the refactoring to pass GeneratorInfo around included this to ensure we were actually setting it correctly, but you are correct: this then immediately got changed with the "AndFreeze" for the next part of the change.